src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -6338,36 +6339,124 @@ subroutine atm_rk_dynamics_substep_finish( state, diag, dynamics_substep, dynami
       call mpas_pool_get_array(state, 'rho_zz', rho_zz_1, 1)
       call mpas_pool_get_array(state, 'rho_zz', rho_zz_2, 2)
 
+      MPAS_ACC_TIMER_START('atm_rk_dynamics_substep_finish [ACC_data_xfer]')
+      !$acc enter data create(ru_save, u_1, rtheta_p_save, theta_m_1, rho_p_save, rw_save, &
+      !$acc                   w_1, rho_zz_1) &
+      !$acc            copyin(ru, u_2, rtheta_p, rho_p, theta_m_2, rho_zz_2, rw, &
+      !$acc                   w_2, ruAvg, wwAvg, ruAvg_split, wwAvg_split, rho_zz_old_split)
+      MPAS_ACC_TIMER_STOP('atm_rk_dynamics_substep_finish [ACC_data_xfer]')
+
+      !$acc kernels
+      theta_m_1(:,cellEnd+1) = 0.0_RKIND
+      !$acc end kernels
+
       inv_dynamics_split = 1.0_RKIND / real(dynamics_split)
       
       if (dynamics_substep < dynamics_split) then
+         !$acc parallel
+         !$acc loop gang worker
+         do iEdge = edgeStart,edgeEnd
+            !$acc loop vector
+            do k = 1,nVertLevels
+               ru_save(k,iEdge) = ru(k,iEdge)
+               u_1(k,iEdge) = u_2(k,iEdge)
+            end do
+         end do
 
-         ru_save(:,edgeStart:edgeEnd) = ru(:,edgeStart:edgeEnd)
-         rw_save(:,cellStart:cellEnd) = rw(:,cellStart:cellEnd)
-         rtheta_p_save(:,cellStart:cellEnd) = rtheta_p(:,cellStart:cellEnd)
-         rho_p_save(:,cellStart:cellEnd) = rho_p(:,cellStart:cellEnd)
 
-         u_1(:,edgeStart:edgeEnd) = u_2(:,edgeStart:edgeEnd)
-         w_1(:,cellStart:cellEnd) = w_2(:,cellStart:cellEnd)
-         theta_m_1(:,cellStart:cellEnd) = theta_m_2(:,cellStart:cellEnd)
-         rho_zz_1(:,cellStart:cellEnd) = rho_zz_2(:,cellStart:cellEnd)
+         !$acc loop gang worker
+         do iCell = cellStart,cellEnd
+            !$acc loop vector
+            do k = 1,nVertLevels
+               rtheta_p_save(k,iCell) = rtheta_p(k,iCell)
+               rho_p_save(k,iCell) = rho_p(k,iCell)
+               theta_m_1(k,iCell) = theta_m_2(k,iCell)
+               rho_zz_1(k,iCell) = rho_zz_2(k,iCell)
+            end do
+         end do
 
+         !$acc loop gang worker
+         do iCell = cellStart,cellEnd
+            !$acc loop vector
+            do k = 1,nVertLevels+1
+               rw_save(k,iCell) = rw(k,iCell)
+               w_1(k,iCell) = w_2(k,iCell)
+            end do
+         end do
+         !$acc end parallel
       end if
 
+
```

Comment:
Can you remove this extra blank line that slipped in?

Reply:
Done.